File: test_flutter/lib/model/Product.dart
class Product {
  int? id;
  String? name;
  int? unitPrice;
  int? stock;
  int? vat;
  String? status;

  Product(
      {this.id, this.name, this.unitPrice, this.stock, this.vat, this.status});

  Product.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    name = json['name'];
    unitPrice = json['unitPrice'];
    stock = json['stock'];
    vat = json['vat'];
    status = json['status'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['name'] = this.name;
    data['unitPrice'] = this.unitPrice;
    data['stock'] = this.stock;
    data['vat'] = this.vat;
    data['status'] = this.status;
    return data;
  }
}

File: crmspring/src/main/java/com/rafiqul/crmspring/entity/Product.java
package com.rafiqul.crmspring.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@Table(name = "product")
@AllArgsConstructor
@NoArgsConstructor
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column(nullable = false)
    private String name;

    private double unitPrice;

    private double stock;

    private float vat;

    @Enumerated(EnumType.STRING)
    private ProductStatus status;

    public enum ProductStatus {
        AVAILABLE,
        OUT_OF_STOCK
    }

}
